pkg/database/mysql/config.go

```diff
@@ -0,0 +1,60 @@
+package mysql
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// Config implements database.Configurator interface and
+// handles the configuration parameters of the mysql database
+type Config struct {
+	Username string // the account username
+	Password string // the account password
+	Hostname string // the hostname where the mysql server is hosted
+	Port     int    // the port on which the server can be accessed
+	Name     string // the database name
+
+	// The database URL can be used to establish the connection without specifying
+	// other credentials
+	URL string
+}
+
+func (t *Config) SetDefaults() {}
+
+func (t *Config) Validate() error {
+	connectionWithParts := !(t.Username == "" || t.Password == "" || t.Hostname == "" || t.Port == 0 || t.Name == "")
+	connectionWithURL := !(t.URL == "")
+
+	if !connectionWithParts && !connectionWithURL {
+		return fmt.Errorf("no method for connection was provided")
+	}
+
+	if t.URL != "" {
+		pu, err := url.Parse(t.URL)
+		if err != nil {
+			return fmt.Errorf("cannot parse connection url: %w", err)
+		}
+
+		// https://github.com/go-sql-driver/mysql#timetime-support
+		if pu.Query().Get("parseTime") != "true" {
```

Comment:
I'd suggest manually appending the `parseTime` query parameter to the URL if it's missing instead of throwing an error if it's not.

In some cases (like Heroku), the environment variables are automatically set (add-ons attachments) and the user cannot modify them.

This will improve a bit the UX and the integration with PaaS clouds.

Reply:
Ok, will do, and then it could be moved to DSN() function.

Reply:
Updated. I suppose in this case there is no point of checking whether **parseTime** is true or not - just always set it.